### PR TITLE
Update `@jupyterlite` packages

### PR DESCRIPTION
**Pull request by jtpio**
- Update to the stable 0.3.0 packages: https://github.com/jupyterlite/jupyterlite/releases/tag/v0.3.0
- Update `jupyterlite-core` and `jupyterlite-sphinx` in `docs/build-environment.yml`

### CODE REVIEW

File: docs/build-environment.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ dependencies:
   - myst-parser
   - pydata-sphinx-theme
   - jupyterlab >=4.0.5,<5
-  - jupyterlite-core >=0.2.0a1,<0.3.0
-  - jupyterlite-sphinx >=0.9.1
+  - jupyterlite-core >=0.3.0,<0.4.0
```
Comment (martinRenou) [inline docs/build-environment.yml:17]: Does it need to be >=0.3?

```suggestion
  - jupyterlite-core >=0.2,<0.4
```

Comment (jtpio) [reply docs/build-environment.yml:17]: For the docs I guess we can use the latest stable?

Comment (jtpio) [reply docs/build-environment.yml:17]: > For the docs I guess we can use the latest stable?

Although we would need to wait for the new version to land on conda-forge.

Comment (jtpio) [reply docs/build-environment.yml]: Changed to `>=0.2,<0.4` for consistency with what is specified in `pyproject.toml`

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**